pkg/3scale/amp/operator/upgrade.go

```diff
@@ -191,21 +192,25 @@ func (u *UpgradeApiManager) upgradeImages() (reconcile.Result, error) {
 		return res, err
 	}
 
-	if !u.apiManager.IsExternalDatabaseEnabled() {
-		res, err = u.upgradeBackendRedisImageStream()
-		if res.Requeue || err != nil {
-			return res, err
-		}
+	upgradeImages := sequentially(
+		when(
+			notExternal(v1alpha1.BackendRedis),
+			(*UpgradeApiManager).upgradeBackendRedisImageStream,
+		),
+		when(
+			notExternal(v1alpha1.SystemRedis),
+			(*UpgradeApiManager).upgradeSystemRedisImageStream,
+		),
+		when(
 
-		res, err = u.upgradeSystemRedisImageStream()
-		if res.Requeue || err != nil {
-			return res, err
-		}
+			notExternal(v1alpha1.SystemDatabase),
+			(*UpgradeApiManager).upgradeSystemDatabaseImageStream,
+		),
+	)
```

Comment:
Hopefully this is not too overcomplicated. The goal is to avoid the alternative chain of `if` blocks, while also making the code more readable (`sequentially, when Backend redis is not external, upgrade backend redis ImageStream, ...`)